pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl

```diff
@@ -9,11 +9,23 @@ The parameter `recursion_dict` is meant to preseve egality
 of converted objects and should never be given by the user.
 """
 
+## Default for conversion:
+## Do not convert Julia types, as long as type matches
+## If you have a GAP list of Julia Tuples, this function
+## enables conversion of that list to a list of type
+## Array{Tuple,1}.
+function gap_to_julia(t::T, x::Any) where T <: Type
+    if ! (typeof(x) <: t)
+        throw(MethodError("Wrong type: expected " * string(t) * ","))
+    end
+    return x
+end
+
 ## Default
 gap_to_julia(::Type{GAPInputType},x::GAPInputType) = x
 gap_to_julia(::Type{Any},         x::GAPInputType) = gap_to_julia(x)
 gap_to_julia(::Type{Any},         x::Any         ) = x
-gap_to_julia(::Any,               x::Nothing     ) = nothing
+gap_to_julia(::T,                 x::Nothing     ) where T <: Type = nothing
 gap_to_julia(::Type{Any},         x::Nothing     ) = nothing
```

Comment:
Are the above three methods still needed now?They are special cases of the new rule, aren't they? (Might still be needed due to its slightly higher method priority, I guess?)

Reply:
Yes, there are still needed.
Removing any one or two of them would either change the behaviour, or make the choice ambigious.
 
In principle, they are subsets of each other, but Julia wants its choice of methods to be unique.